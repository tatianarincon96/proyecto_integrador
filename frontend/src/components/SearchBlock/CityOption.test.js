import React from 'react';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import '@testing-library/jest-dom'
import CityOption from './CityOption';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({mode:'deep'}))
let wrapper = shallow(<CityOption />);

beforeEach(() => {
    wrapper = shallow(<CityOption />);
})

describe('Pruebas en <CityOption />', () => {  
    it("Deberia mostrar <CityOption/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('<CityOption /> should render the city in props', () => {
        const city = 'Buenos Aires';
        const jsxCityOption = shallow(<CityOption city={city}/>);
        expect(jsxCityOption).toMatchSnapshot();
    });
    it('<CityOption /> should render the country in props', () => {
        const country = 'Argentina';
        const jsxCityOption = shallow(<CityOption country={country}/>);
        expect(jsxCityOption).toMatchSnapshot();
    });
    it('should render one <CityOption>', () => {
        expect(wrapper.find('h3')).toHaveLength(2);
    });
});