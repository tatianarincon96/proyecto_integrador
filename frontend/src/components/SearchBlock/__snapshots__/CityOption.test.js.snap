// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CityOption /> <CityOption /> should render the city in props 1`] = `
<Fragment>
  <h3
    className="titleCity"
    onClick={[Function]}
  >
    Buenos Aires
    , 
  </h3>
  <h3
    className="titleCountry"
    onClick={[Function]}
  />
</Fragment>
`;

exports[`Pruebas en <CityOption /> <CityOption /> should render the country in props 1`] = `
<Fragment>
  <h3
    className="titleCity"
    onClick={[Function]}
  >
    , 
  </h3>
  <h3
    className="titleCountry"
    onClick={[Function]}
  >
    Argentina
  </h3>
</Fragment>
`;

exports[`Pruebas en <CityOption /> Deberia mostrar <CityOption/> correctamente 1`] = `
<Fragment>
  <h3
    className="titleCity"
    onClick={[Function]}
  >
    , 
  </h3>
  <h3
    className="titleCountry"
    onClick={[Function]}
  />
</Fragment>
`;
